geode-assembly/src/distributedTest/java/org/apache/geode/session/tests/TomcatTest.java

```diff
@@ -0,0 +1,267 @@
+      int operatingContainer)
+      throws IOException, URISyntaxException {
+    String key = getKey(client, attributes);
+    int maxLength = 100;
+    String value = getRandomVarChar(client, attributes, maxLength);
+    attributes.put(key, value);
+    verifySets(client, attributes, operatingContainer, key, value);
+  }
+
+  private void verifySets(Client client, HashMap<String, String> attributes, int operatingContainer,
+      String key, String value) throws IOException, URISyntaxException {
+    int whichContainer =
+        isCachingClient() ? operatingContainer : random.nextInt(numberOfContainers);
+    client.setPort(Integer.parseInt(manager.getContainerPort(whichContainer)));
+    Client.Response resp = client.set(key, value);
+    if (isCachingClient()) {
+      getKeyValueDataOnOperatingClient(client, key, value, resp.getSessionCookie(),
+          operatingContainer);
+      verifySessionGets(client, attributes, resp, operatingContainer);
+    } else {
+      getKeyValueDataOnAllClients(client, key, value, resp.getSessionCookie());
+      verifySessionGets(client, attributes, resp);
+    }
+  }
+
+  private void doSetsAndGetWithUpdate(Client client, HashMap<String, String> attributes,
+      int operatingContainer)
+      throws IOException, URISyntaxException {
+    String key = getKey(client, attributes);
+    if (random.nextInt(10) == 1) {
+      // do update on an existing attribute
+      key = "key";
+    }
+    int maxLength = 100;
+    String value = getRandomVarChar(client, attributes, maxLength);
+    attributes.put(key, value);
+    verifySets(client, attributes, operatingContainer, key, value);
+  }
+
+  protected String getKey(Client client, Map attributes) {
+    return "key" + (attributes.size() + 1) + "_" + client.getCookie();
+  }
+
+  protected String getRandomVarChar(Client client, Map attributes, int length) {
+    if (length == 0) {
+      return "";
+    }
+    StringBuffer buffer = new StringBuffer();
+    buffer.append("value" + (attributes.size() + 1) + "_" + client.getCookie());
+    int randomLength = random.nextInt(length) + 1;
+
+    int sp = ' ';
+    int tilde = '~';
+    for (int j = 0; j < randomLength; j++) {
+      buffer.append((char) (random.nextInt(tilde - sp) + sp));
+    }
+    return buffer.toString();
+  }
+
+  private void verifySessionGets(Client client, HashMap<String, String> attributes,
+      Client.Response resp) throws IOException, URISyntaxException {
+    verifySessionGets(client, attributes, resp, -1);
+  }
+
+  private void verifySessionGets(Client client, HashMap<String, String> attributes,
+      Client.Response resp, int operatingContainer) throws IOException, URISyntaxException {
+    int count = 0;
+    int whichOne = random.nextInt(attributes.size());
+    Iterator iterator = attributes.entrySet().iterator();
+    while (iterator.hasNext()) {
+      Map.Entry entry = (Map.Entry) iterator.next();
+      if (count == whichOne) {
```

Comment:
Why not verify all the entries in `attributes`?

Reply:
All attributes are verified at the end of test. No need to verify the same attributes after each operation (basically verify same attributes again.) Test will just randomly check one of the attributes and verify all at the end of the test.